core/templates/dev/head/components/top_navigation_bar/TopNavigationBarDirective.js

```diff
@@ -95,13 +95,114 @@ oppia.directive('topNavigationBar', [function() {
         }
 
         $scope.windowIsNarrow = windowDimensionsService.isWindowNarrow();
+        var currentWindowWidth = windowDimensionsService.getWidth();
+        $scope.navElementsVisibilityStatus = {
+          I18N_TOPNAV_DONATE: true,
+          I18N_TOPNAV_ABOUT: true,
+          I18N_CREATE_EXPLORATION_CREATE: true,
+          I18N_TOPNAV_LIBRARY: true
+        };
+
         windowDimensionsService.registerOnResizeHook(function() {
           $scope.windowIsNarrow = windowDimensionsService.isWindowNarrow();
           $scope.$apply();
           // Close the sidebar, if necessary.
           SidebarStatusService.closeSidebar();
+
+          // If the window is resized larger, try displaying the hidden elements
+          if (currentWindowWidth < windowDimensionsService.getWidth()) {
+            $.each($scope.navElementsVisibilityStatus, function(e, visible) {
+              if (!visible) {
+                $scope.navElementsVisibilityStatus[e] = true;
+              }
+            });
+          }
+          currentWindowWidth = windowDimensionsService.getWidth();
+          oppiaDebouncer.debounce(truncateNavbar, 500)();
```

Comment:
Re the debouncer -- great catch! I took a look, and I think there are a couple of errors in the oppiaDebouncer.debounce():
- Can you try removing the "last > 0" part of the if statement?
- Could you move the timeout, context, args, timestamp and result vars above the general return statement for the factory (so that they become, essentially, private vars in the factory)? Previously they were local variables, and I think that that was what was causing the problem, because the timeout would always be empty.

That said, we could have problems if there are multiple debouncers in a page (and in fact this could happen in the library page since we have the search debouncer). It seems like Angular has its own [debouncer](https://docs.angularjs.org/api/ng/directive/ngModelOptions); do you think we could use that for the search box and reserve this one for the navbar resizing? (We should put a big warning on the debouncer to say that this is only used for the navbar, or move the logic into the TopNavigationBarDirective if it's not being used elsewhere.)